packages/local-environment/src/domain.ts

```diff
@@ -41,27 +54,28 @@ export class NomadDomain {
       watcher: signer,
       recoveryManager: signer,
     });
+
   }
 
-  addNomadEnv(e: NomadEnv) {
+  addNomadEnv(e: NomadEnv): void {
     this.nomadEnv = e;
   }
 
   get name(): string {
     return this.network.name;
   }
 
-  ensureAgents(metricsPort = 9090) {
-    if (!this.agents) this.agents = new Agents(this, metricsPort, this.nomadEnv);
+  ensureAgents(metricsPort = 9090): void {
+    if (!this.agents) this.agents = new Agents(this, metricsPort, this.docker, this.nomadEnv);
   }
 
-  connectNetwork(d: NomadDomain) {
+  connectNetwork(d: NomadDomain): void {
     if (!this.connections().includes(d.network.name))
       this.connectedNetworks.push(d);
   }
 
-  async isAgentsUp(): Promise<boolean> {
-    return await this.agents!.isAllUp();
+  async isAgentsUp(): Promise<boolean | undefined> {
+    return await this.agents?.isAllUp();
```

Comment:
If `this.agents` is not defined, does that mean that they're still spinning up?

Reply:
No this just checks docker status for all agents.

Reply:
Just realized we might wanna use `areAgentsUp` as opposed to `isAgentsUp` lol